server/plugin.go

```diff
@@ -312,20 +313,22 @@ func (p *Plugin) start(isRestart bool) {
 		p.syncUserJob = job
 	}
 
-	checkCredentialsJob, err := cluster.Schedule(
-		p.API,
-		checkCredentialsJobName,
-		cluster.MakeWaitForRoundedInterval(24*time.Hour),
-		p.checkCredentials,
-	)
-	if err != nil {
-		p.API.LogError("error in scheduling the check credentials job", "error", err)
-		return
-	}
-	p.checkCredentialsJob = checkCredentialsJob
+	if !p.getConfiguration().DisableCheckCredentials {
+		checkCredentialsJob, jobErr := cluster.Schedule(
+			p.API,
+			checkCredentialsJobName,
+			cluster.MakeWaitForRoundedInterval(24*time.Hour),
+			p.checkCredentials,
+		)
+		if jobErr != nil {
+			p.API.LogError("error in scheduling the check credentials job", "error", jobErr)
+			return
+		}
+		p.checkCredentialsJob = checkCredentialsJob
 
-	// Run the job above right away so we immediately populate metrics.
-	p.checkCredentials()
+		// Run the job above right away so we immediately populate metrics.
+		p.checkCredentials()
+	}
```

Comment:
Allow the tests not to run this job and trigger unexpected mock calls to the MS Teams Graph API. There's an opportunity to improve things here in the future.